components/brave_ads/content/browser/search_result_ad/search_result_ad_service.cc

```diff
@@ -40,11 +40,10 @@ void SearchResultAdService::MaybeRetrieveSearchResultAd(
     content::RenderFrameHost* render_frame_host,
     SessionID tab_id,
     bool should_trigger_viewed_event) {
-  DCHECK(ads_service_);
   DCHECK(render_frame_host);
   DCHECK(tab_id.is_valid());
 
-  if (!should_trigger_viewed_event || !ads_service_->IsEnabled() ||
+  if (!ads_service_->IsEnabled() ||
```

Comment:
ads service can be null
```AdsService* AdsServiceFactory::GetForProfile(Profile* profile) {
  if (!brave::IsRegularProfile(profile)) {
    return nullptr;
  }

  return static_cast<AdsService*>(
      GetInstance()->GetServiceForBrowserContext(profile, true));
}```

Reply:
and SearchResultAdService won't be created for incognito browser, see factory code:
https://github.com/brave/brave-core/blob/93a3167ac3722808b044d06d46b7ab727382aa9d/browser/brave_ads/search_result_ad/search_result_ad_service_factory.cc

Reply:
then why did you remove the DCHECK for `ads_service_`?

Reply:
there is already the DCHECK in `SearchResultAdService` constructor, so I think that it is enough to have one DCHECK, as `ads_service_` is set only in constructor.

Reply:
maybe we could make `raw_ptr<AdsService> ads_service_ = nullptr;` const. But I do notice we also `set ads_service_` in `SetAdsServiceForTesting`.

Reply:
Agreed that `ads_service_` can be changed in tests, so need to return `DCHECK` back